### DISCUSSION TITLE
Cannot sent eMail via SMTP

### DISCUSSION THREAD

**Discussion by gymnae** [ente-io/ente#3365] [Q&A] [answered]
Thank you for your great project and putting effort into enabling self-hosting. This is a lot of effort and not directly leading to revenue. So I appreciate the effort. 

I set up self-hosting in an LXC Linux container in a Proxmox hypervisor. It's working apart from SMTP sending. 
I verified the credentials were correct, and I entered them into museum.yaml without quotes, seeing that the email.go code adds those. Nevertheless, I also tried adding single and double quotes to the variable inputs.

The error message which appears after some minutes is the following:
```
ERRO[8723]handler.go:47 Error Request failed                                error= --- at /home/ente/ente/server/pkg/api/user.go:40 (UserHandler.SendOTT) ---
 --- at /home/ente/ente/server/pkg/controller/user/userauth.go:129 (UserController.SendEmailOTT) ---
 --- at /home/ente/ente/server/pkg/controller/user/userauth.go:313 (emailOTT) ---
 --- at /home/ente/ente/server/pkg/utils/email/email.go:109 (sendViaSMTP) ---
Caused by: EOF req_id=ser_xDOeSa8PcoinvikRzPOlGU user_id=0
```
Sending an email with the same credentials via another application works.

**Comment by mnvr** [comment]
Thank you for your kind words.

There are myriad reasons this could be failing, and it is most likely that the credentials are being mangled somewhere. I can't suggest anything better than debugging this using plain ol' printfs. You can build a museum from source (you can leave your existing one setup untouched, you can build it separately, and connect to it only to get it to debug the reason the smtp send is failing). Once you have it building from source, you can start adding printfs - you essentially want to see what values are reaching this point - https://github.com/ente-io/ente/blob/405c0c343ff0fdf822c5f2bc8cbab358d10ff38e/server/pkg/utils/email/email.go#L47

**Reply by mnvr** [reply]
(Also, you're likely aware, but just in case I'll mention: If you just want the OTP and don't want to debug the reason, you can read it off the logs - https://help.ente.io/self-hosting/faq/otp)

**Reply by gymnae** [reply]
Hi, thanks for your quick reply. I’d like working email in case commenting or new upload notifications are added in the future.
I’m a total newb with coding. I’m running ente in an VM without docker, so i start with the go command. What would i need to do to use printfs?

**Reply by gymnae** [reply, marked as answer]
Got it figured out, was also able to print ouf debug messages.
The issue was: I was trying with port 465, which is SSL/TLS - but only port 587 with STARTTLS is supported
